core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -101,19 +104,24 @@ public static BlockInStream createLocalBlockInStream(long blockId, long blockSiz
    * @return the {@link BlockInStream} created
    */
   // TODO(peis): Use options idiom (ALLUXIO-2579).
-  public static BlockInStream createRemoteBlockInStream(long blockId, long blockSize,
+  public static BlockInStream createNettyBlockInStream(long blockId, long blockSize,
       WorkerNetAddress workerNetAddress, FileSystemContext context, InStreamOptions options) {
     Closer closer = Closer.create();
     try {
       BlockWorkerClient blockWorkerClient =
           closer.register(context.createBlockWorkerClient(workerNetAddress));
       LockBlockResource lockBlockResource =
           closer.register(blockWorkerClient.lockBlock(blockId, LockBlockOptions.defaults()));
+      SocketAddress address;
+      if (NettyClient.isDomainSocketSupported(workerNetAddress)) {
```

Comment:
I can optionally pass the result this call from StreamFactory. But i thought that this call is cheap enough (a string comparison) and didn't do that. Let me know if you prefer more params to be passed.